solidity_contracts/src/CairoPrecompiles/MulticallCairoCounterCaller.sol

```diff
@@ -0,0 +1,209 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0 <0.9.0;
+
+using MulticallCairoLib for uint256;
+
+/// @notice A contract that performs various call types to the Kakarot MulticallCairo precompile.
+/// @dev Only meant to test the MulticallCairo precompile when called from a Solidity Contract.
+contract MulticallCairoCounterCaller {
+    /// @dev The cairo contract to call
+    uint256 immutable cairoCounter;
+
+    /// @dev The cairo function selector to call - `inc`
+    uint256 constant FUNCTION_SELECTOR_INC = uint256(keccak256("inc")) % 2 ** 250;
+
+    /// @dev The cairo function selector to call - `set_counter`
+    uint256 constant FUNCTION_SELECTOR_SET_COUNTER = uint256(keccak256("set_counter")) % 2 ** 250;
+
+    constructor(uint256 cairoContractAddress) {
+        cairoCounter = cairoContractAddress;
+    }
+
+    /// @notice Calls the Cairo contract to increment its internal counter
+    function incrementCairoCounter() external {
+        cairoCounter.callCairo("inc");
+    }
+
+    /// @notice Calls the Cairo contract to increment its internal counter in a batch of multiple calls
+    function incrementCairoCounterBatch(uint32 n_calls) external {
+        uint256[][] memory calls = new uint256[][](n_calls);
+        for (uint32 i = 0; i < n_calls; i++) {
+            calls[i] = new uint256[](3);
+            calls[i][0] = cairoCounter;
+            calls[i][1] = FUNCTION_SELECTOR_INC;
+            calls[i][2] = 0; // Length of the empty calldata array
+                // No data because len = 0
+        }
+        MulticallCairoLib.batchCallCairo(calls);
+    }
+
+    /// @notice Calls the Cairo contract to increment its internal counter
+    /// @dev The delegatecall preserves the caller's context, so the caller's address will
+    /// be the caller of this function.
+    /// @dev Should always fail, as MulticallCairo does not support delegatecalls.
+    function incrementCairoCounterDelegatecall() external {
+        cairoCounter.delegatecallCairo("inc");
+    }
+
+    /// @notice Calls the Cairo contract to increment its internal counter
+    /// @dev Called with a regular call, the caller's address will be this contract's address
+    /// @dev Should always fail, as MulticallCairo does not support callcode.
+    function incrementCairoCounterCallcode() external {
+        cairoCounter.callcodeCairo("inc");
+    }
+}
+
+library MulticallCairoLib {
+    /// @dev The Batch Cairo precompile contract's address.
+    address constant BATCH_CAIRO_PRECOMPILE_ADDRESS = 0x0000000000000000000000000000000000075003;
+
+    function concatenateBytes(bytes memory a, bytes memory b) public pure returns (bytes memory) {
```

Comment:
After tinkering with [chisel](https://book.getfoundry.sh/chisel/),  you could replace this by `abi.encodePacked(a,b)` or `a.concat(b)`

Reply:
bytes.concat(a,b) is the right syntax, I couldn't get it to work, thanks